nodl_to_policy/_common/_profile.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from lxml import etree
+try:
+    import importlib.resources as importlib_resources
+except ModuleNotFoundError:
+    import importlib_resources  # type: ignore
+
+
+# For now, only use `node.xml`, since NoDL does not describe Lifecycle nodes
+def common_profile() -> etree.ElementTree:
+    return _get_profile('node.xml')
+
+
+def common_subscribe_topics() -> List:
+    return _get_items_by_role('topics', 'subscribe')
+
+
+def common_publish_topics() -> List:
+    return _get_items_by_role('topics', 'publish')
+
+
+def common_reply_services() -> List:
+    return _get_items_by_role('services', 'reply')
+
+
+def common_request_services() -> List:
+    return _get_items_by_role('services', 'request')
+
+
+def _get_profile(filename: str) -> etree.ElementTree:
+    # Parses the `node.xml` or the `lifecycle_node.xml` file
+    # The latter filename is WIP, check TODO above in source
+    with importlib_resources.path('nodl_to_policy._common', filename) as path:
+        profile = etree.parse(str(path))
+    profile.xinclude()
+    return profile
+
+
+def _get_items_by_role(item_type: str, role: str) -> List:
```

Comment:
What happens in this function if `item_type` or `role` are the empty string?

Reply:
Good point. If `item_type` is an empty string, it returns an empty list for the common profile - although really it's looking for an element with an empty tag, which is not what we want to happen. The latter throws an error because of a predicate error - again, not something we want to happen.

As such, I think it's best to return an empty list whenever either of the two arguments are empty. Something like:

```suggestion
def _get_items_by_role(item_type: str, role: str) -> List:
    if not item_type or not role:
        return []
```

Reply:
Addressed in c11f6e7.